server/routes/routerRequestHandlers/retrieveGoals.ts

```diff
@@ -0,0 +1,51 @@
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+import { EducationAndWorkPlanService } from '../../services'
+import logger from '../../../logger'
+import asyncMiddleware from '../../middleware/asyncMiddleware'
+import GoalStatusValue from '../../enums/goalStatusValue'
+
+/**
+ *  Middleware function that returns a Request handler function to retrieve the goals by status from EducationAndWorkPlanService and store in res.locals
+ */
+const retrieveGoals = (educationAndWorkPlanService: EducationAndWorkPlanService): RequestHandler => {
```

Comment:
I think the `educationAndWorkPlanService` already handles whether the API client returned an error or not, so I dont think you need to worry about that here.
I think all you want here is essentially what was previously in the controller. Something like this:

```
const retrieveGoals = (educationAndWorkPlanService: EducationAndWorkPlanService): RequestHandler => {
  return asyncMiddleware(async (req: Request, res: Response, next: NextFunction) => {
    const { prisonNumber } = req.params
    const status: GoalStatusValue = req.query.status as GoalStatusValue

    // Retrieve the goals by status and store in res.locals
    res.locals.goals = await educationAndWorkPlanService.getGoalsByStatus(prisonNumber, status, req.user.username)

    next()
  })
}
```

I think thats all you need 🤔